### ISSUE TITLE
How to attach event to X button on max modal, for old app-bridge version

### ISSUE THREAD

**Issue by iamphonghg** [Shopify/shopify-app-bridge#297]
My app is still using @shopify/app-bridge and @shopify/app-bridge-react version 3.4.3; and use Fullscreen.create(app) for Fullscreen mode.

Shopify automatically changes fullscreen mode to max modal. And when pressing the X button on the modal, it only turns off the modal but does not exit Fullscreen mode. So after pressing the X button, you still have to continue pressing the Back button (On Fullscreen bar) (perform action fullscreen.dispatch(Fullscreen.Action.EXIT)) to exit fullscreen mode.

So is there a way to attach an event to the X button, or simply hide it?

**Comment by Zetxus** [comment]
Just to chime in - I've also found that the 3.x.x version of the app-bridge is broken by the modal changes. We're on 3.5.0 and we've started to urgently migrate to 4.0.0 to avoid the same issue - fullscreen is now a variant=max modal but the content is rendered behind the modal as well, so essentially you have to either click X and Exit the fullscreen or click Exit twice, as the Fullscreen bar is still present. 

Other than that to answer @iamphonghg, I think `onHide` in the new Modal API will do what you want it to do. It's called whenever the x is pressed or the user clicks outside of the modal (on the max variant that would be the tiny gap between the modal's borders and the app's iframe.

**Comment by iamphonghg** [comment]
@Zetxus Of course the new modal api can fix it. But I think it's too risky to upgrade a core package just for a small problem.

**Comment by shatishdesai202** [comment]
Hi @charlesdobson ,

looks like the React solution is not working.

   ```
const unsubscribeExit = fullscreen.subscribe(Fullscreen.Action.EXIT, () => {
    setEmbeddedFullScreen(false);
  });

  useEffect(() => {
    return () => {
      unsubscribeExit();
    };
  }, [unsubscribeExit]);
```

**Comment by shatishdesai202** [comment]
@iamphonghg,
 Yes, correct.

**Comment by mrvan2851** [comment]
```
  const unsubscribeExit = app.subscribe(Fullscreen.Action.EXIT, data => {});
  ```
this way still working

**Comment by shatishdesai202** [comment]
Hi @mrvan2851 @iamphonghg ,
I have tried and tested it, it's working fine for me,

`  const unsubscribeExit = app.subscribe(Fullscreen.Action.EXIT, data => {});`

**Comment by darrynten** [comment]
@mrvan2851 @iamphonghg

please note, you must subscribe on the app now for it to work, not on the fullscreen instance like before

**incorrect**

```
fullscreen.subscribe(Fullscreen.Action.EXIT, () => {})
```

**correct**

```
app.subscribe(Fullscreen.Action.EXIT, () => {})
```

Once we changed my listener away from fullscreen and onto the app object the X button on the modal started triggering this event and we're able to gain control of the exit process again

**Comment by iamphonghg** [comment]
Hi @darrynten, thanks so much, could you see why I can't do it?

On the page that use fullscreen I have:
` useEffect(() => {
    fullscreen.dispatch(enterAction);
    return () => {
      fullscreen.dispatch(exitAction);
    };
  }, []);`

I tried putting your code in AppBridgeProvider, after the app is created but it doesn't work.
`const app = createApp(appBridgeConfig);
 app.subscribe(Fullscreen.Action.EXIT, () => {});`

**Comment by iamphonghg** [comment]
Oh, I think I have found the problem.

**Comment by hartherbert** [comment]
> app.subscribe(Fullscreen.Action.EXIT, () => {})

While this works if you just want to exit the fullscreen, it does not prevent navigation when the contextual save bar is present.
It closes the max modal but the fullscreen is still present and you are on the same page, if the navigation is canceled.

This is just half a solution for the given force max modal problem.

**Comment by hartherbert** [comment]
Yeah I already found a way with this but we still can't get "Built for Shopify" verified, since we cannot setup the max modal buttons. It is required to use those buttons and not the contextual savebar.

**Comment by darrynten** [comment]
I don't think any existing app with a modal more complex than a basic contact form is ever going to qualify again given that non-trivial code no longer works in the modal context
